### PR TITLE
Internalized `Endpoint.Prepare`

### PR DESCRIPTION
**Pull request by danielmarbach**
Made `Endpoint.Prepare` internal. There is currently no use case for `Prepare` @andreasohlund @timbussmann and I could come up with. This makes the public API smaller.

## Documentation

https://github.com/Particular/docs.particular.net/pull/1344

### CODE REVIEW

File: src/NServiceBus.Core/Endpoint.cs [hunk 1]
```diff
@@ -11,9 +11,8 @@ public static class Endpoint
         /// Creates a new initializable endpoint based on the provided configuration.
         /// </summary>
         /// <param name="configuration">Configuration.</param>
-        public static IInitializableEndpoint Prepare(EndpointConfiguration configuration)
+        static IInitializableEndpoint Prepare(EndpointConfiguration configuration)
```
Comment (andreasohlund) [inline src/NServiceBus.Core/Endpoint.cs]: Why not remove the entire method?

**Comment by andreasohlund** [comment]
Checked again and the only usage I can find is this doco https://github.com/Particular/docs.particular.net/blob/93dc9cf5ca98561731055a5017ae93f8580531f3/Snippets/Snippets_6/Hosting.cs

**Comment by danielmarbach** [comment]
@andreasohlund I'm updating the doco as we speak

File: src/NServiceBus.Core/Endpoint.cs (hunk 1 above)
Comment (danielmarbach) [inline src/NServiceBus.Core/Endpoint.cs]: Why not leave it here?

Comment (andreasohlund) [inline src/NServiceBus.Core/Endpoint.cs]: A lot of code just to do `configuration.Build();` ?

Comment (andreasohlund) [inline src/NServiceBus.Core/Endpoint.cs]: And that means you can remove `IInitializableEndpoint` as well?

Deleted code == good code

**Comment by danielmarbach** [comment]
@Particular/nservicebus-maintainers please review

**Comment by andreasohlund** [comment]
Remove https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/IInitializableEndpoint.cs ?

**Comment by andreasohlund** [comment]
Lol, realtime reviews ;)

**Comment by danielmarbach** [comment]
@andreasohlund Done. I decided to leave the separation internally between `InitializableEndpoint` and `StartableEndpoint`

**Comment by danielmarbach** [comment]
Ping @bording

**Comment by bording** [comment]
This looks good! I know it was mentioned in #3471 to consider removing `Create` from the public API also. 

I can see that it's currently being used by the the Acceptance tests, but other than that, what is the use case for keeping `Create` around?

**Comment by andreasohlund** [comment]
Create is used by all our hosts so I think we have to keep it

A smart client might also want to create but start on demand.

Merging this since @bording gave the thumbs up and we can deal with `Create` separately if needed

**Pull request merged**